src/utils/api/helper.ts

```diff
@@ -593,6 +595,58 @@ export const calculateFees = async ({
   )();
 };
 
+export const proceedToLogin = async ({
+  recaptchaRef,
+  onError,
+  onResponse,
+}: {
+  recaptchaRef: ReCAPTCHA;
+  onError: (e: string) => void;
+  onResponse: (r: any) => void;
+}) => {
+  const token = await callRecaptcha(recaptchaRef, true);
```

Comment:
why not include thin in chain? something like 
```suggestion
  pipe(
    fromNullable(token),
    fromOption(() => {
      onError(ErrorsType.GENERIC_ERROR);
      return 'Token missing';
    }),
    chain(...)
```

Reply:
Unfortunately, requesting a major refactoring of `callRecaptcha`, which could impact other parts of the checkout.